### DISCUSSION TITLE
When will Lazer mapping be ready??

### DISCUSSION THREAD

**Discussion by StalkyBoi43815** [ppy/osu#27158] [Q&A]
I really like the Lazer editor, and I find the stable editor to be bloated, ugly and hard to use. Mapping on Lazer is quite the _breeze_. However, it is very incomplete and empty, and one cannot even upload a damn map! And, as an Arch user btw, I don't really like running stable on Wine. I mean, it works perfectly but I just don't like it. I can't wait for mapping on Lazer, and I am excited for the future of osu! (Lazer). So, when will the Lazer editor be ready? I know PP is the main focus and demand now, but still. Peppy pls fix.

**Comment by cihe13375** [comment]
The majority of mapping work can be done on lazer already; notable deficiencies include map upload, hitsounding (custom samples, slider tails, etc.), storyboard, distance snap (it has some weirdness, the most dominant one maybe https://github.com/ppy/osu/issues/21347), and the "Songs" folder like stable (to interact with tools like mapset verifier).

For map upload, the current wordaround is to export the finished map to stable: map in lazer -> export .osz (file - export - for compatibility .osz) -> import to stable ->upload.

I think the biggest reason preventing lazer map upload is that lazer and stable use different file formats (that's why we need an export "for compatibility"), and if lazer maps are uploaded directly they would not be correctly parsed by stable.

**Comment by bdach** [comment]
we do not give estimates, you'll just have to be patient, sorry. we're working on pp / infrastructure / gameplay concerns right now. well aware beatmap submission is a huge hole in the editor right now, but there are priorities.

**Reply by StalkyBoi43815** [reply]
I understand as most players probably haven't ever even touched the editor. I've never developed a game so I can only imagine how hard and time-consuming it must be. Just looking forward to the future of Lazer mapping.
